ui/redux/selectors/collections.js

```diff
@@ -40,6 +40,26 @@ export const selectUnpublishedCollectionsList = createSelector(
   (unpublishedCollections) => Object.keys(unpublishedCollections || {})
 );
 
+export const selectCollectionSavedForId = (state: State, id: string) => {
+  const savedIds = selectSavedCollectionIds(state);
+  return savedIds.includes(id);
+};
+
+export const selectSavedCollections = createSelector(
+  selectSavedCollectionIds,
+  (state) => (id) => selectCollectionForId(state, id),
+  (savedIds, collectionForId) => {
```

Comment:
I haven't loaded to fully try it, but I think this will not memoize correctly because the second selector, `(state) => (id) => selectCollectionForId(state, id),`, will always return a new inline function, so `selectSavedCollections` will always re-evaluate.

But I also see the problem .. you would then need to bring along every selector encapsulated by `selectCollectionForId` here..

```
  selectSavedCollectionIds,
  // plus
  selectBuiltinCollections,
  selectResolvedCollections,
  selectMyUnpublishedCollections,
  selectMyEditedCollections,
  selectPendingCollections,
  selectCurrentQueueList,
  (... ) => {}
```

If cannot find a simpler solution, I guess not cache it for now (use simple function), since this won't be caching correctly.

Reply:
do you think this is any different? https://github.com/OdyseeTeam/odysee-frontend/commit/d32e0a84e36fbd7290a2571767a63e35cce6df59

Reply:
Yeah, saving into redux is a good alternative if the selector got too complex.

- :small_blue_diamond: `savedCollections: { source: 'collections', property: 'savedIds' },` this got very confusing because there is already a state in redux called `savedCollections`.  It's also confusing when viewing the wallet because it's not a collection map like the rest, but an array of collection IDs.
- :grey_question: Can you summarize the purpose of `withEdits`?
  - `collectionsWithEdits` and `editedCollections` mean the same thing at first glance of the wallet, so that's confusing.
  - For the most part, they seem to duplicates, even for the timestamps.
  - When I clicked "clear edits", that should just restore the collection back into the published state and remove it from `editedCollections`, right? But the wallet still contains the `editedCollections` entry.  What does this scenario represent?
    - <img width="1205" alt="image" src="https://user-images.githubusercontent.com/64950861/181157992-f4326ea0-58fa-47ef-8824-e445e09046d2.png">

Reply:
- Improved it a bit, could've used the `resolved` object all this time for saved. The saved collections are stored as an array of ids since they are public playlists, they need to be resolved to store the current state (they can be updated, deleted, etc)
- `withEdits` now `update` the only purpose is to delete it from `edited` but still keep track of when was last updated, since `edited` only applies to public collections, it's safe to delete the `edited` entry when edits are cleared since it will then select the `resolved` entry (without edits)

Reply:
It much cleaner now, thanks.

- [ ] :small_blue_diamond: `updatedCollections` is still writing the entire collection.
  - In `SharedData`, it is defined as `updatedCollections: UpdatedCollection`.  I think that's good to not bloat the wallet.
  - In the wallet, it is written as a full `Collection`.  I think need to transform the data in `sharedStateFilters`.
- [ ] :grey_question: If I cleared an edit, and later edit again + really publish it, does it make sense for the `updatedCollections` entry to still be there?  It should probably just use the claim's timestamp at that point.
  - Back in [previous comment](https://github.com/OdyseeTeam/odysee-frontend/pull/1877#discussion_r930599284), I was confused because I expected "Last Updated At" to just revert to the claim's timestamp when edits are cleared, so `withEdits | updated` felt unnecessary -- simpler model.  But I can also see the benefit of listing when it was last touched, even if  reverted.

Reply:
Fixed `updatedCollections` and looks like removing edited was a thing so now it should delete updated instead (should be even better than before now since you keep your edited version, let's say you used LBRY desktop to publish the collection but forgot you had added some other items on odysee, the edits shouldn't be cleared by a most recent version, but I didn't test this)